code/script.js

```diff
@@ -262,82 +417,163 @@ const selectQuestion = () => {
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
-    }
-  } else if (group === 'other') {
-    // Similar to the one above
+// Show the correct alert message for different categories
+const { attribute, category, value } = currentQuestion
+
+if (category === 'accessories') {
+  if (keep) {
+    alert(
+      `You are correct, the person wears ${attribute}! Keep all that wears ${attribute}.`
+    )
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
-    } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+    alert(
+      `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}.`
+    )
   }
-
+} else if (category === 'hair color') {
+  if (keep) {
+    alert(`Sure thing, the person has ${value} hair! Keep all that has ${value} hair.`
+    )
+  } else {
+    alert(
+    `Nope, the person doesn't have ${value} hair. Remove all that has ${value} hair.`
+    )
+  }
+} else if (category === 'eye color') {
+  if (keep) {
+    alert(
+      `Correctamundo, the person has ${value} eyes! Keep all that has ${value} eyes.`
+    )
+  } else {
+    alert(
+      `Sorry, the person doesn't have ${value} eyes. Remove all that has ${value} eyes.`
+      )
+  }
+} else if (category === 'other') {
+  if (keep) {
+    alert(
+      `Affirmative, the person has a ${attribute}. Keep all that has a ${attribute}.`
+    )
+} else {
+    alert(
+      `Negative, the person doesn't have a ${attribute}. Remove all that has a ${attribute}.`
+    )
+  } 
+}
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
+  addGuessesMade(keep)
   // Invoke a function to redraw the board with the remaining people.
 }
 
+// Adds the current guess to the list of guesses made
+const addGuessesMade = (keep) => {
+  if (keep) {
+    guessesMade.innerHTML += `
+  <span class="right-guess"><img class="yes-or-no" src="./images/yes.svg">${currentQuestion.text}</span>
+  `
+  }
+  else {
+    guessesMade.innerHTML += `
+    <span class="wrong-guess"><img class="yes-or-no" src="./images/no.svg">${currentQuestion.text}</span>
+    `
+  }
+}
```

Comment:
❗ This function should be removed.
Reasons for this:
- It causes an error in console
![image](https://user-images.githubusercontent.com/3659568/107963871-fa551180-6fa8-11eb-928b-09b574d960d6.png)
- It appears that the function serves no purpose (and I suspect this was used previously but was later replaced by **numberOfGuesses** variable on line  450 & 451.)

Reply:
I tried to do a function to show what guesses you had already made, but couldn't make it work. Got a bit stressed in the end, so I forgott stuff in the code... :/ Same thing goes for unused classes further down!

Reply:
The counter stops counting if I comment this out. :/